main.cpp

```diff
@@ -39,29 +39,251 @@ send me a DM to check your pull request
+        void loadWavetable(int wavetableToLoad, float loadTimeOffset);
+    };
 
+    int getNextFrequencyInSequence(float nextFrequency)
+    {
+        std::cout << "The next frequency is: " << nextFrequency << ".\n" << std::endl;
+
+        std::cout << "Waveshape is: " << waveShape << ". Volume Level is: " << volumeLevel << ".\n" << std::endl;
+        return{};
+    }
+
+    float fadeOutVolume(WavetableOscillator osc, float incrementAmount)
+    {
+        for(float i = osc.volumeLevel; i >= 0; i -= incrementAmount)
+        {
+            osc.volumeLevel -= incrementAmount;
+            std::cout << "Current oscillator volume is: " << osc.volumeLevel << "." << std::endl;
+            if(osc.volumeLevel <= 0)
+            {
+                std::cout << "Finished fading out!\n" << std::endl;
+                return osc.volumeLevel;
+            }
+        }
+        
+        return osc.volumeLevel;
+    }
+};
 /*
  copied UDT 3:
  */
+struct Reverb
+{
+    int decayTime;
+    float roomSize;
+    double inputVolume;
+    double outputVolume;
+    double dryWet;
+
+    Reverb() : decayTime(1), roomSize(3.5f), inputVolume(1.0), outputVolume(1.0), dryWet(1.0) 
+    {
+        std::cout << "Spawning Reverb object!\n" << std::endl;
+    }
+
+    ~Reverb()
+    {
+        std::cout << "Destroying Reverb object!\n" << std::endl;
+    }
+
+    int setNextReverbSettings(Reverb newVerbSettings);
+
+    double setNextDryWet(Reverb targetVerb, double nextDryWetValue)
+    {
+        double epsilon = 0.0001;
+
+        std::cout << "\nAdjusting reverb dryWet!" << std::endl;
+        double incrementPolarity;
+
+        if(targetVerb.dryWet < nextDryWetValue) incrementPolarity = 0.1;
+        else incrementPolarity = -0.1;
+
+        if(nextDryWetValue > 1) 
+        {
+            std::cout << "dryWet cannot be higher than 1.0!" << std::endl;
+            return dryWet;
+        }
+
+        for(double instancedDryWet = targetVerb.dryWet; std::abs(instancedDryWet - nextDryWetValue) > epsilon; instancedDryWet += incrementPolarity)
+        {
+            std::cout << "Current dryWet is: " << instancedDryWet << "." << std::endl;
+
+            if(instancedDryWet <= nextDryWetValue)
+            {
+                std::cout << "Finished! New dryWet is: " << instancedDryWet << ".\n" << std::endl;
+                return dryWet;
+            }
+        }
+        return dryWet;
+    }
+};
+
+int Reverb::setNextReverbSettings(Reverb newVerbSettings)
+{
+    Reverb oldVerbSettings;
+    if(newVerbSettings.decayTime == oldVerbSettings.decayTime)
+    {
+        decayTime = 0;
+        std::cout << "Setting reverb decay time to: " << decayTime << ".\n" << std::endl;
+    }
+    else
+    {
+        newVerbSettings.decayTime = oldVerbSettings.decayTime;
+        std::cout << "Setting reverb decay time to old verb settings: (" << newVerbSettings.decayTime << ").\n\n" << std::endl;
+    }
```

Comment:
`oldVerbSettings` shouldn't be here.  You should be using your member variables of the calling instance. 
You're already doing that on 241. 
- [ ] revise function to use member variables of the calling instance.